metricflow/plan_conversion/node_processor.py

```diff
@@ -272,6 +345,49 @@ def _add_time_range_constraint(
                 processed_nodes.append(source_node)
         return processed_nodes
 
+    def _add_where_constraint(
+        self,
+        source_nodes: Sequence[DataflowPlanNode],
+        where_filter_specs: Sequence[WhereFilterSpec],
+        enabled_element_types: FrozenSet[LinkableElementType],
+    ) -> Sequence[DataflowPlanNode]:
+        """Processes where filter specs and evaluates their fitness for pushdown against the provided node set."""
+        eligible_filter_specs_by_model: Dict[SemanticModelReference, Sequence[WhereFilterSpec]] = {}
+        for spec in where_filter_specs:
+            semantic_models = set(element.semantic_model_origin for element in spec.linkable_elements)
+            invalid_element_types = [
+                element for element in spec.linkable_elements if element.element_type not in enabled_element_types
+            ]
+            if len(semantic_models) == 1 and len(invalid_element_types) == 0:
```

Comment:
Ok, so we only push down filters if ALL the filtered elements are eligible element types. Is that because we don't know if this is an `AND` or an `OR` filter at this point?